src/types.d.ts

```diff
@@ -61,9 +61,11 @@ export type EngineExecutionSummary = {
 	violationCount: number;
 };
 
+export type RecombinedData = string | {columns: string[]; rows: {[key: string]: string|number}[]};
```

Comment:
Defining this new type makes changes in `RuleResultRecombinator` and `RunOutputProcessor` much simpler.